test/functional/p2p_tx_download.py

```diff
@@ -241,6 +243,30 @@ def test_spurious_notfound(self):
         self.log.info('Check that spurious notfound is ignored')
         self.nodes[0].p2ps[0].send_message(msg_notfound(vec=[CInv(MSG_TX, 1)]))
 
+    def test_rejected_tx_request_logic(self):
+        self.log.info('Check that rejected tx is not requested again')
+        minrelaytxfee = self.nodes[0].getmempoolinfo()['minrelaytxfee']
+        peer = self.nodes[0].add_p2p_connection(TestP2PConn())
+        low_fee_tx = self.wallet.create_self_transfer(fee_rate=Decimal("0.9")*minrelaytxfee)
+        with self.nodes[0].assert_debug_log(expected_msgs=['was not accepted: min relay fee not met']):
```

Comment:
Btw if you wanted to hit the problem in https://github.com/bitcoin/bitcoin/pull/28970#discussion_r1553911167, we would want the error to be "mempool min fee not met" (would need `fill_mempool`)